packages/x-data-grid/src/hooks/features/columnHeaders/useGridColumnHeaders.tsx

```diff
@@ -66,7 +66,7 @@ export interface GetHeadersParams {
 
 const SpaceFiller = styled('div')({
   /* GridRootStyles conflict */
-  '&&': {
+  '&&&': {
```

Comment:
I don't know why, but style order changes on color mode change, and `&&` is not sufficient anymore:

https://github.com/mui/mui-x/assets/13808724/e3d89e48-b489-4c4c-9fe9-04af732347b1